fault.c

```diff
@@ -391,12 +770,19 @@ int faults_check(struct kibosh_faults *faults, const char *path, const char *op)
 {
     struct kibosh_fault_base **iter;
 
+    int ret = 0;
+
     for (iter = faults->list; *iter; iter++) {
-        int ret = kibosh_fault_base_check(*iter, path, op);
-        if (ret)
-            return ret;
+        int err_code = kibosh_fault_base_check(*iter, path, op);
+        if (err_code) {
+            // get the smallest err_codes
+            // i.e. when unreadable and read_corrupt faults both present, we want to execute unreadable
```

Comment:
This comment does not seem very accurate. Could you elaborate a bit on what this piece of code is doing?

Reply:
The reason I added this is to handle the case where two read faults are injected for the same path and file_type. The error code for unreadable faults are from 1 to 122, and error code (mode) for corrupt faults are greater than 1000. So if both faults are presented, unreadable will be applied. Without the check, Kibosh would arbitrarily inject the first fault it finds, so the result is not deterministic.